tests/integration-tests/utils.py

```diff
@@ -116,3 +119,72 @@ def delete_s3_bucket(bucket_name, region):
         bucket.delete()
     except boto3.client("s3").exceptions.NoSuchBucket:
         pass
+
+
+def set_credentials(region, credential_arg):
+    """
+    Set credentials for boto3 clients and cli commands
+
+    :param region: region of the bucket
+    :param credential_arg: credential list
+    """
+    if credential_arg:
+        # credentials = dict { region1: (endpoint1, arn1, external_id1),
+        #                      region2: (endpoint2, arn2, external_id2),
+        #                      [...],
+        #                    }
+        credentials = {
+            region: (endpoint, arn, external_id)
+            for region, endpoint, arn, external_id in [
+                tuple(credential_tuple.strip().split(","))
+                for credential_tuple in credential_arg
+                if credential_tuple.strip()
+            ]
+        }
+
+        if region in credentials:
+            aws_credentials = _retrieve_sts_credential(credentials, region)
+
+            # Set credential for all boto3 client
+            boto3.setup_default_session(
+                aws_access_key_id=aws_credentials["AccessKeyId"],
+                aws_secret_access_key=aws_credentials["SecretAccessKey"],
+                aws_session_token=aws_credentials["SessionToken"],
+            )
+
+            # Set credential for all cli command e.g. pcluster create
+            os.environ["AWS_ACCESS_KEY_ID"] = aws_credentials["AccessKeyId"]
+            os.environ["AWS_SECRET_ACCESS_KEY"] = aws_credentials["SecretAccessKey"]
+            os.environ["AWS_SESSION_TOKEN"] = aws_credentials["SessionToken"]
+
+
+def _retrieve_sts_credential(credentials, region):
+    credential_endpoint = credentials.get(region)[0]
```

Comment:
minor you could do something like:
```
a, b, c = credentials.get(region)
```

also you could consider passing already the credentials for the specific region to the function